cmake/ccache.cmake

```diff
@@ -5,8 +5,8 @@ if(WITH_CCACHE)
   find_program (CCACHE_FOUND ccache)
   if(CCACHE_FOUND)
     message(STATUS "Building with ccache: ${CCACHE_FOUND}, CCACHE_DIR=$ENV{CCACHE_DIR}")
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
+    set(CMAKE_C_COMPILER_LAUNCHER ccache)
+    set(CMAKE_CXX_COMPILER_LAUNCHER ccache)
```

Comment:
Using [`CMAKE_<LANG>_COMPILER_LAUNCHER`](https://cmake.org/cmake/help/latest/prop_tgt/LANG_COMPILER_LAUNCHER.html) because it's not a property but a variable and can be easily passed to the third-party dependencies through CMake (also because it's the preferred way of doing things). Passing `ccache` to `RULE_LAUNCH_LINK` makes no difference so here we set only the compiler launcher.